tests/integration/conftest.py

```diff
@@ -389,3 +389,73 @@ def _load_json_schema(path):
 
     with open(path, mode="r") as handler:
         return json.load(handler)
+
+
+@pytest.fixture
+def pre_registered(shell):
+    """
+    A fixture to install subscription manager and pre-register the system prior to the convert2rhel run.
+    """
+    assert shell("yum install -y subscription-manager").returncode == 0
+    # Download the SSL certificate
+    shell("curl --create-dirs -o /etc/rhsm/ca/redhat-uep.pem https://ftp.redhat.com/redhat/convert2rhel/redhat-uep.pem")
+    # Register the system
+    assert (
+        shell(
+            "subscription-manager register --serverurl {} --username {} --password {}".format(
+                env.str("RHSM_SERVER_URL"), env.str("RHSM_USERNAME"), env.str("RHSM_PASSWORD")
+            )
+        ).returncode
+        == 0
+    )
+
+    assert shell("subscription-manager attach --pool {}".format(env.str("RHSM_POOL"))).returncode == 0
+
+    rhsm_uuid_command = "subscription-manager identity | grep identity"
+
+    uuid_raw_output = shell(rhsm_uuid_command).output
+
+    # The `subscription-manager identity | grep identity` command returns
+    # system identity: <UUID>, we need to store just the system UUID
+    original_registration_uuid = uuid_raw_output.split(":")[1].strip()
+
+    yield
+
+    # For some scenarios we do not pre-register the system, therefore we do not have the original UUID
+    # and do not need to verify it stays the same
+    if "C2R_TESTS_CHECK_RHSM_UUID_MATCH" in os.environ:
```

Comment:
I would add an `else` branch where if the `C2R_TESTS_CHECK_RHSM_UUID_MATCH` is set to `0` we check that the UUID does not match

Reply:
There is no such case, where the UUID won't match.
It either matches, or there is nothing to check against.
* pre-registered system without passing credentials to the convert2rhel - we have UUID prior to the conversion, and we do not re-register, so the UUID will be same either after conversion, or after rollback
* pre-registered system with credentials provided to the convert2rhel - we have UUID prior to the conversion, but convert2rhel will re-register, and potentially unregister during rollback. Therefore, we do not have the post rollback system UUID
** we don't cover pre-registered system with provided credentials to convert2rhel doing the full conversion.
* unregistered system without passing credentials to convert2rhel will fail the registration action and exit the conversion - there is no UUID prior, nor post convert2rhel run

Reply:
You are right, there is no such case for that. Thanks for the explanation!